### PR TITLE
Login access errors

### PR DESCRIPTION
**Pull request by FedericoSanucciWX**
### Description
The app validates the state of the network, the response of the api service and show toast messages. Progress bars appears when login button is clicked to inform the status.
Note: toast messages only informs description of the error (old images attached)

### Screenshot
Shutter_001
Shutter_002
Shutter_003

### Trello
[https://trello.com/c/bNa1q2NU/10-u2-backend-login-errores](url)

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/training/ui/training/login/LoginFragment.java [hunk 1]
```diff
@@ -30,6 +35,8 @@
     private TextInputEditText mEmailTxt;
     private TextInputEditText mPassTxt;
 
+    private ProgressDialog pDialog;
```
Comment (dylan-muszel) [inline app/src/main/java/ar/com/wolox/android/training/ui/training/login/LoginFragment.java]: All the member attributes should be named by `m...`.

File: app/src/main/java/ar/com/wolox/android/training/ui/training/login/LoginFragment.java [hunk 2]
```diff
@@ -160,6 +172,34 @@ public void showMainScreen() {
         startActivity(intent);
     }
 
+    @Override
+    public void showProgressDialog() {
+        if (pDialog != null) {
```
Comment (dylan-muszel) [inline app/src/main/java/ar/com/wolox/android/training/ui/training/login/LoginFragment.java]: It's not necessary, or is it?

File: app/src/main/java/ar/com/wolox/android/training/ui/training/login/LoginFragment.java [hunk 3]
```diff
@@ -160,6 +172,34 @@ public void showMainScreen() {
...
+            pDialog.setMessage(getString(R.string.login_service_request));
+            pDialog.show();
+        }
+    }
+
+    @Override
+    public void hideProgressDialog() {
+        if (pDialog != null) {
```
Comment (dylan-muszel) [inline app/src/main/java/ar/com/wolox/android/training/ui/training/login/LoginFragment.java]: Same as above.

File: app/src/main/java/ar/com/wolox/android/training/ui/training/login/LoginPresenter.java [hunk 4]
```diff
@@ -93,37 +93,46 @@ void onTermsAndConditionClicked() {
...
-                if (response.isSuccessful()) {
-                    try {
-                        if (response.body() != null) {
-                            List<User> users = response.body();
-                            if (users.size() < 1) {
-                                getView().showInvalidCredentialsError();
-                            } else if (users.size() > 1) {
-                                getView().showMultiplesCredentialsError();
+        getView().showProgressDialog();
+        if (!getView().isNetworkAvailable()) {
```
Comment (dylan-muszel) [inline app/src/main/java/ar/com/wolox/android/training/ui/training/login/LoginPresenter.java:97]: Perfect! ❤️

**Review by dylan-muszel — CHANGES_REQUESTED** [review]
Some minor comments, but it's a really good work. I love the way you resolve to use MVP with the network check.

**Comment by FedericoSanucciWX** [comment]
Attributes renamed and removed unnecessary statements. Thanks for the patience!!

**Review by dylan-muszel — APPROVED** [review]
Perfect! Remember to change the PR base branch!

**Pull request closed without merging**